login/token_src.go

```diff
@@ -224,11 +230,11 @@ func (s *OIDCTokenSource) newToken() (*oidc.Token, error) {
 		extra.Set("nonce", nonce)
 	}
 
-	ctx, cancel := context.WithTimeout(s.ctx, 1*time.Minute)
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

Comment:
this is interesting one. I guess most of the user context will have stricter timeout than 1 minute here...  Maybe we should not base on user context here?

Reply:
Don't have much context here :D should I base it on context.Background()?